credentials/sts/sts.go

```diff
@@ -192,15 +190,8 @@ type httpDoer interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-func makeHTTPClient(roots *x509.CertPool) httpDoer {
-	return &http.Client{
-		Timeout: stsRequestTimeout,
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs: roots,
-			},
-		},
-	}
+func makeHTTPClient(client *http.Client) httpDoer {
```

Comment:
Is this function required?  All it does is return its parameter.

Reply:
right. its an artifact just for supporting the preexisting testcases.  i'm not sure how to untangle those cleanly (nor would it be something to look at in this PR).  If you have a tested suggestion on how to do that, i'd hear it but i'd pin refactoring the existing tests to use the actual grpc and sts server added to in this PR for a later task

Reply:
This function appears to have been used by tests exactly to implement the functionality you just exported as first-class functionality - to set a custom HTTP client for the credentials to use.  Just pass it in via the `Options.HTTPClient` field instead, and delete this and `makeHTTPDoer`?

It looks like to do this, the field type of `Options.HTTPClient` will need to be changed to `httpDoer` instead of `*http.Client` (exported, obviously).  This seems like a good idea anyway as it means users could pass other HTTP client implementations, and we only need the Do method.

Reply:
dunno.  i looked at this and the comment and can't figure it out.  I'm not gonna spend time rejiggering and making up for a odd test case artifact is used in the main code.  this FR is a nice to have for me to demo cert-bound sts creds but not necessary.    i can leave this PRs code here incase someone here wants to refactor this